File: Erlang/lists1.erl
-module(lists1).
-export([myLast/1, myButLast/1, min/1]).

myLast([X]) -> X;
myLast([_|T]) -> myLast(T).

myButLast([_]) -> false;
myButLast([X,_]) -> X;
myButLast([H|T]) -> myButLast(T).

min([H|T]) ->
    min(H, T).
min(M, []) ->
    M;
min(M, [H|T]) when M < H ->
    min(M, T);
min(_M, [H|T]) ->
    min(H, T).

File: Haskell/lists1.hs
myLast :: [a] -> a
myLast [] = error "No end of empty lists!"
myLast [x] = x
myLast (_:xs) = myLast xs

myButLast :: [a] -> a
myButLast = last . init

insertAt :: a -> [a] -> Int -> [a]
insertAt y xs n = take (n-1) xs ++ [y] ++ drop (n-1) xs

isPalindrome :: (Eq a) => [a] -> Bool
isPalindrome xs = xs == reverse xs
